File: UtilitiesVB/Base Functionality Extensions/ExtToChar.vb
Imports System.Runtime.CompilerServices
Imports System.Windows.Forms

Public Module ExtToChar
    <Extension()>
    Public Function ToChar(ByVal key As Keys) As Char
        Dim c As Char = vbNullChar

        If (key >= Keys.A) AndAlso (key <= Keys.Z) Then
            c = ChrW(AscW("a"c) + (key - Keys.A))
        ElseIf (key >= Keys.D0) AndAlso (key <= Keys.D9) Then
            c = ChrW(AscW("0"c) + (key - Keys.D0))
        End If

        Return c
    End Function
End Module

File: UtilitiesVB/Base Functionality Extensions/ExtToChar.cs
using System.Windows.Forms;
using Microsoft.VisualBasic;
using Microsoft.VisualBasic.CompilerServices;

namespace UtilitiesVB
{

    public static class ExtToChar
    {
        public static char ToChar(this Keys key)
        {
            char c = Conversions.ToChar(Constants.vbNullChar);

            if (key >= Keys.A && key <= Keys.Z)
            {
                c = Strings.ChrW(Strings.AscW('a') + ((int)key - (int)Keys.A));
            }
            else if (key >= Keys.D0 && key <= Keys.D9)
            {
                c = Strings.ChrW(Strings.AscW('0') + ((int)key - (int)Keys.D0));
            }

            return c;
        }
    }
}